tests/unit/test_svs.cpp

```diff
@@ -2095,6 +2111,52 @@ TYPED_TEST(SVSTest, rangeQuery) {
     VecSimIndex_Free(index);
 }
 
+TYPED_TEST(SVSTest, joinSearchParams) {
+
+    auto qbits = TypeParam::get_quant_bits();
+    bool is_two_level_lvq = [=]() {
+        switch (qbits) {
+        case VecSimSvsQuant_4x4:
+        case VecSimSvsQuant_4x8:
+        case VecSimSvsQuant_4x8_LeanVec:
+        case VecSimSvsQuant_8x8_LeanVec:
+            return true;
+        default:
+            return false;
+        }
+    }();
+
+    // only change window size = 100
+    // sp should have window size = 100, buffer capacity = 100 or
+    // 150 if Two-level LVQ is enabled
+    SVSRuntimeParams svsRuntimeParams = {.windowSize = 100};
+    auto query_params = CreateQueryParams(svsRuntimeParams);
+    auto sp = svs_details::joinSearchParams({}, &query_params, is_two_level_lvq);
+    ASSERT_EQ(sp.buffer_config_.get_search_window_size(), svsRuntimeParams.windowSize);
+    ASSERT_EQ(sp.buffer_config_.get_total_capacity(),
+              (is_two_level_lvq) ? static_cast<size_t>(1.5 * svsRuntimeParams.windowSize)
+                                 : svsRuntimeParams.windowSize);
+
+    // change both window size and buffer capacity
+    // sp should change based on runtime parameters
+    svsRuntimeParams.windowSize = 200;
+    svsRuntimeParams.bufferCapacity = 300;
+    query_params = CreateQueryParams(svsRuntimeParams);
+    sp = svs_details::joinSearchParams({}, &query_params, is_two_level_lvq);
+    ASSERT_EQ(sp.buffer_config_.get_search_window_size(), svsRuntimeParams.windowSize);
+    ASSERT_EQ(sp.buffer_config_.get_total_capacity(), svsRuntimeParams.bufferCapacity);
+
+    // only change buffer capacity = 100
+    // buffer capacity is changed only if window size is changed
+    // sp2 should be the same as previous sp
+    svsRuntimeParams.windowSize = 0;
+    svsRuntimeParams.bufferCapacity = 100;
```

Comment:
Not sure I understood this case, if `svsRuntimeParams.windowSize = 0`, then how come `sp2.buffer_config_.get_search_window_size() = 200`? Shouldn't it be the default value (10)?